### PR TITLE
feature/Add example for go

### PR DESCRIPTION
**Pull request by Djenchik**

### CODE REVIEW

File: examples/go/example.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	log.Print(response.Message)
+}
+
+func getPrivateKey() (*rsa.PrivateKey, error) {
+	filePath := filepath.Join("keys", "example_private")
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, errors.Annotate(err, "opening file failed")
+	}
+	defer func() error {
```
Comment (Nazaroff88) [inline examples/go/example.go:63]:
```suggestion
	}
	
	defer func() error {
```

File: examples/go/example.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+		if err = file.Close(); err != nil {
+			return errors.Annotate(err, "closing file failed")
+		}
+		return nil
```
Comment (Nazaroff88) [inline examples/go/example.go:67]:
```suggestion
		}
		
		return nil
```

**Review by Oxxyg33n — APPROVED** [review]

File: examples/go/main.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+		if !claims.VerifyExpiresAt(time.Now().UTC().Unix(), true) {
+			return nil, errors.New("invalid token expiration time")
+		}
+
+		if claims["sub"] != subject {
+			return nil, errors.New("invalid subject")
+		}
+
+		if token.Header["alg"] == nil || token.Header["alg"] != jwt.SigningMethodRS256.Alg() {
```
Comment (d0tfile) [inline examples/go/main.go:173]: I think nil pointer check is not needed here

File: examples/go/main.go [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+			return nil, errors.New("invalid token signing algorithm")
+		}
+
+		expectedRBH, err := calculateRequestBodyHash(result.ResponseBody)
+		if err != nil {
+			return nil, errors.Annotate(err, "hashing httpResponse body failed")
```
Comment (d0tfile) [inline examples/go/main.go:179]:
```suggestion
			return nil, errors.Annotate(err, "hashing response body failed")
```

File: examples/go/main.go [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+		return nil, errors.Annotate(err, "writing sha256 hash failed")
+	}
+
+	return hash.Sum(nil), nil
+}
+
+func decodePEMBlock(block []byte) ([]byte, error) {
+	decodedKey, _ := pem.Decode(block)
+	if decodedKey == nil {
+		log.Fatal("decoding block failed")
```
Comment (d0tfile) [inline examples/go/main.go:251]: `return nil, errors.Annotate(...)`

**Review by Oxxyg33n — DISMISSED** [review]

**Review by d0tfile — APPROVED** [review]

**Review by Oxxyg33n — APPROVED** [review]

**Pull request merged**